src/app/blog/frost-threshold-signatures/page.mdx

```diff
@@ -0,0 +1,346 @@
+
+With [pkarr] we have a great mechanism to publish information about keypairs. We are running a dns server, and we can also use the bittorrent [mainline](https://en.wikipedia.org/wiki/Mainline_DHT) DHT for a fully peer to peer mechanism to publish and resolve pkarr packets.
+
+Other recent protocols such as [nostr] are also using ed25519 keypairs.
+
+## How to keep the keys safe
+
+A problem that frequently comes up when using a keypair to control access to an identity or a resource is how to keep the private key safe.
+
+Some keypairs are ephemeral and don't need to be safeguarded much.
+
+Some will have significant security implications from the start (e.g. a keypair associated with access to a crypto wallet).
+
+And some will initially be of low value, but might grow in value over time (e.g. a social media account).
+
+In most cases, there is a constant conflict between the need to keep the keys safe and the need to constantly access the private key for signing messages.
+
+## Existing solutions
+
+### Local file system
+
+The default way to store a private key is to just store it in a hidden directory in your local file system. While this is not extremely secure, it is still highly preferable to not using encryption at all. In many scenarios, e.g. device loss or theft, this is perfectly fine for low to medium value keypairs.
+
+### Secure key storage
+
+Most modern hardware supports secure storage for private keys. However, access to such secure storage locations is highly platform dependent. It also only works for a limited set of cryptographic primitives, which might not include EdDSA. More fundamentally, while secure storage makes the key relatively inaccessible, it does not protect against key loss. It also does not provide a mechanism for revocation.
+
+### Delegation schemes
+
+With the tools of public key cryptography you can come up with delegation schemes where a rarely used master key is used to delegate to a more frequently used keypair that can be revoked using the master key. This is a very complex topic that would require its own series of blog posts.
+
+## Threshold signatures
+
+I was vaguely aware that something like threshold signatures exist. This is - very roughly speaking - a scheme where you split the private key into multiple parts called *shares*, and need a certain number of these shares to sign a message. Since the shares never have to be in one place, this provides safety in case a single share gets compromised or lost.
+
+What I did not know however is that there exist threshold signature schemes such as FROST that work with Ed25519 such that generated signatures are *fully compatible* with normal Ed25519 signatures. So you can sign a message with a threshold signature scheme and then validate the signature as usual using the ed public key.
```

Comment:
Well, I'd say that backwards compatibility is not really the "new" thing here. It's still true that it is surprising!
The reason for that is mostly that with FROST we have an MPC protocol to generate a signature, without having to reconstruct the secret anywhere.
The fact that *that* signature is ed25519-backwards-compatible was surprising to me.
But reconstructing secrets to sign stuff in essentially a backwards-compatible way was done since shamir secret sharing was invented.

Reply:
I don't claim that this is new, just that I did not know it... Probably hard to believe, but I also did not know about shamir secret sharing. However, given the fact that it requires you to collect shares in one place I don't find it that useful for this use case.

WDYT, just add a sentence namedropping shamir secret sharing and call it a day?

> Compared to Shamir Secret Sharing, the FROST scheme does not require shares to be in one place to sign.